### PR TITLE
Add slug field to category, product, collection & page details

### PR DESCRIPTION
**Pull request by mmarkusik**
[SALEOR-1021](https://app.clickup.com/t/2549495/SALEOR-1021)

- Also changed the seo form subtitle to only show when form is collapsed, confirmed with @pwgryglak 

✨ After changes:

Screenshot 2020-09-21 at 15 06 31
Screenshot 2020-09-21 at 15 18 00
Screenshot 2020-09-21 at 15 14 29

### Pull Request Checklist

1. [x] Type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1021 As a User I want to change slug for Product so that I can adjust the SEO based to my needs](https://app.clickup.com/t/2549495/SALEOR-1021)

File: src/components/SeoForm/SeoForm.tsx [hunk 1]
```diff
@@ -89,6 +94,7 @@ const SeoForm: React.FC<SeoFormProps> = props => {
   const intl = useIntl();
   const [expanded, setExpansionStatus] = React.useState(false);
   const toggleExpansion = () => setExpansionStatus(!expanded);
+  const shouldDisplayHelperText = () => helperText && !expanded;
```
Comment (dominik-zeglen) [inline src/components/SeoForm/SeoForm.tsx:97]: It doesn't have to be a function 🤷‍♂️

```suggestion
  const shouldDisplayHelperText = helperText && !expanded;
```

**Review by dominik-zeglen — APPROVED** [review]

File: src/components/SeoForm/SeoForm.tsx (hunk 1 above)
Comment (mmarkusik) [reply src/components/SeoForm/SeoForm.tsx]: Tru 😅 Changed

**Review by orzechdev — APPROVED** [review]

**Review by AlicjaSzu — APPROVED** [review]

**Comment by krzysztofwolski** [comment]
@mmarkusik seems to be a bug:
1. goto any product,  https://feat-add-slug-to-seo-form.dashboard.saleor.rocks/products/UHJvZHVjdDo3Mg%3D%3D
2. modified slug 
3. slug is properly saved
4. empty field, click on save. I'm getting error without any visible error msg

**Review by dominik-zeglen — APPROVED** [review]

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]
Actually no, slug field needs to be able to display error 🤷‍♂️

**Comment by mlamperski** [comment]
@mmarkusik maybe bug:
Special characters on slug field:
1. goto any product, https://feat-add-slug-to-seo-form.dashboard.saleor.rocks/products/UHJvZHVjdDo3Mw%3D%3D
2. modified slug
3. slug is properly saved
3. modify field using special characters (!@#$%^^&*  etc..).

IS: system allows to put special characters into slug field
SHOULD: system should not allow special characters from input

File: src/components/SeoForm/SeoForm.tsx [hunk 2]
```diff
@@ -78,18 +105,69 @@ const SeoForm: React.FC<SeoFormProps> = props => {
...
+      return intl.formatMessage({
+        defaultMessage: DEFAULT_INPUT_MESSAGE
+      });
+    }
+
+    if (error) {
+      return getSlugErrorText();
+    }
+
+    return "";
```
Comment (tomaszszymanski129) [inline src/components/SeoForm/SeoForm.tsx:151]:
```suggestion
    return error ? getSlugErrorText() : "";
```

File: src/components/SeoForm/SeoForm.tsx [hunk 3]
```diff
@@ -78,18 +95,65 @@ const SeoForm: React.FC<SeoFormProps> = props => {
...
+  const getSlugErrorMessage = () => {
+    const error = getError(SeoField.slug);
+    const { __typename: type } = error;
+
+    return type === "ProductError"
+      ? getProductErrorMessage(error as ProductErrorFragment, intl)
+      : getPageErrorMessage(error as PageErrorFragment, intl);
+  };
+
+  const handleSlugChange = event => {
```
Comment (dominik-zeglen) [inline src/components/SeoForm/SeoForm.tsx:147]: Untyped function argument

File: src/components/SeoForm/SeoForm.tsx [hunk 4]
```diff
@@ -78,18 +95,65 @@ const SeoForm: React.FC<SeoFormProps> = props => {
...
+    const { value } = event.target;
+
+    if (value === "" || SLUG_REGEX.test(value)) {
+      onChange(event);
+    }
+  };
+
+  const getError = fieldName => getFieldError(errors, fieldName);
```
Comment (dominik-zeglen) [inline src/components/SeoForm/SeoForm.tsx:155]: Untyped function argument

File: src/pages/components/PageDetailsPage/PageDetailsPage.tsx [hunk 5]
```diff
@@ -56,6 +56,7 @@ const PageDetailsPage: React.FC<PageDetailsPageProps> = ({
 }) => {
   const intl = useIntl();
   const localizeDate = useDateLocalize();
+  const pageExists = page === null;
```
Comment (dominik-zeglen) [inline src/pages/components/PageDetailsPage/PageDetailsPage.tsx:59]: `pageExists` is true if page doesn't exist

File: src/components/SeoForm/SeoForm.tsx [hunk 6]
```diff
@@ -168,11 +261,8 @@ const SeoForm: React.FC<SeoFormProps> = props => {
                   )}
                 </div>
               }
-              helperText={intl.formatMessage({
-                defaultMessage:
-                  "If empty, the preview shows what will be autogenerated."
-              })}
-              value={description ? description.slice(0, 299) : undefined}
+              helperText={intl.formatMessage(seoFieldMessage)}
```
Comment (dominik-zeglen) [inline src/components/SeoForm/SeoForm.tsx:264]: If `getError(SeoField.description) === true`, this input will be red but won't display any message

File: src/components/SeoForm/SeoForm.tsx [hunk 7]
```diff
@@ -65,9 +78,13 @@ interface SeoFormProps {
   description?: string;
   descriptionPlaceholder: string;
   disabled?: boolean;
+  errors?: Array<PageErrorFragment | ProductErrorFragment>;
   loading?: boolean;
   helperText?: string;
+  isCreating?: boolean;
```
Comment (dominik-zeglen) [inline src/components/SeoForm/SeoForm.tsx:84]: I think the better name would be `allowEmptySlug`, it's more understandable from the component's perspective

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

**Review by orzechdev — CHANGES_REQUESTED** [review]

File: src/components/SeoForm/SeoForm.tsx (hunk 6 above)
Comment (mmarkusik) [reply src/components/SeoForm/SeoForm.tsx:264]: Ok, removing until we get errors for those fields from backend

**Review by orzechdev — APPROVED** [review]

**Review by dominik-zeglen — APPROVED** [review]

**Pull request merged**